src/main/java/com/google/devtools/build/lib/analysis/BaseRuleClasses.java

```diff
@@ -343,10 +346,23 @@ public Metadata getMetadata() {
     return builder.add(attr("name", STRING).nonconfigurable("Rule name"));
   }
 
-  public static RuleClass.Builder execPropertiesAttribute(RuleClass.Builder builder)
+  /**
+   * Adds an {@code exec_properties} attribute of type {@code STRING_DICT} if the rule does not
+   * already have one.
+   */
+  public static void addOrOverrideExecPropertiesAttribute(RuleClass.Builder builder)
```

Comment:
I think the previous name is fine. It's low-enough use that reading the source or doc to find out what happens when the attribute already exists is fine.